src/Components/WebAssembly/Server/src/WebAssemblyRazorComponentsBuilderExtensions.cs

```diff
@@ -41,6 +43,7 @@ public static IRazorComponentsBuilder AddInteractiveWebAssemblyComponents(this I
     public static IRazorComponentsBuilder AddAuthenticationStateSerialization(this IRazorComponentsBuilder builder, Action<AuthenticationStateSerializationOptions>? configure = null)
     {
         builder.Services.TryAddEnumerable(ServiceDescriptor.Scoped<IHostEnvironmentAuthenticationStateProvider, AuthenticationStateSerializer>());
+        RegisterPersistentComponentStateServiceCollectionExtensions.AddPersistentServiceRegistration<AuthenticationStateProvider>(builder.Services, RenderMode.InteractiveWebAssembly);
```

Comment:
No, it should be:
```suggestion

builder.Services.TryAddEnumerable(ServiceDescriptor.Scoped<AuthenticationStateProvider, AuthenticationStateSerializer>());
builder.Services.TryAddScoped(sp => (AuthenticationStateSerializer)sp.GetRequiredService<AuthenticationStateProvider>());
RegisterPersistentComponentStateServiceCollectionExtensions.AddPersistentServiceRegistration<AuthenticationStateProvider>(builder.Services, RenderMode.InteractiveAuto);
```

Reply:
Wrong, please stick to my suggestion. I used `AuthenticationStateProvider` as a base class, so you should update the implementation of `AuthenticationStateSerializer` accordingly.